catalyst_voices/lib/widgets/menu/voices_node_menu.dart

```diff
@@ -18,10 +18,58 @@ final class VoicesNodeMenuItem extends Equatable {
   List<Object?> get props => [id, label];
 }
 
+final class VoicesNodeMenuData extends Equatable {
+  final int? selectedItemId;
+  final bool isExpanded;
+
+  const VoicesNodeMenuData({
+    this.selectedItemId,
+    this.isExpanded = false,
+  });
+
+  VoicesNodeMenuData copyWith({
+    int? selectedItemId,
+    bool? isExpanded,
+  }) {
+    return VoicesNodeMenuData(
+      selectedItemId: selectedItemId ?? this.selectedItemId,
+      isExpanded: isExpanded ?? this.isExpanded,
+    );
+  }
+
+  VoicesNodeMenuData clearSelection() {
+    return VoicesNodeMenuData(
+      selectedItemId: null,
+      isExpanded: isExpanded,
+    );
+  }
+
+  @override
+  List<Object?> get props => [
+        selectedItemId,
+        isExpanded,
+      ];
+}
+
+final class VoicesNodeMenuController extends ValueNotifier<VoicesNodeMenuData> {
+  set selected(int? newValue) {
+    value = newValue != null
+        ? value.copyWith(selectedItemId: newValue)
+        : value.clearSelection();
+  }
+
+  set isExpanded(bool newValue) {
```

Comment:
The project is following dart style convention, not flutter style convention. Getter / setter should be under constructor. Fields should be above the constructor.